ocaml/gencert/selfcert.ml

```diff
@@ -14,23 +14,28 @@
 
 module C = Cmdliner
 module Rsa = Mirage_crypto_pk.Rsa
-open Rresult
+module UX = Xapi_stdext_unix.Unixext
+open Rresult (* introduces >>= >>| and R *)
+
+(** initialize the random number generator at program startup when this
+module is loaded. *)
+let () = Mirage_crypto_rng_unix.initialize ()
 
 let defer f = Fun.protect ~finally:f
 
 (** [write_cert] writes a PEM file to [path]. It attempts to do that
   atomically by writing to a temporary file in the same directory first and
   renaming the file at the end *)
 let write_certs path key cert =
-  let delimit = "\n" in
+  let delim = "\n" in
   let temp_dir = Filename.dirname path in
   let tmp = Filename.temp_file ~temp_dir "certify-" ".tmp" in
   try
     let fd = Unix.openfile tmp [Unix.O_WRONLY] 0o600 in
```

Comment:
This should be 0o400, like in https://github.com/xapi-project/xen-api/blob/master/ocaml/gencert/lib.ml#L165

I'm not happy that there are two functions for writing a PKCS12 to disk, could you change this function so the installation one in lib.ml uses this one? i would also prefer if `UX.atomic_write_to_file` was used as well to avoid  duplicated functionality.